src/Symfony/Component/Yaml/Tag/TagResolver.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Yaml\Tag;
+
+use Symfony\Component\Yaml\Yaml;
+use Symfony\Component\Yaml\Exception\UnsupportedTagException;
+
+/**
+ * @author Guilhem N. <egetick@gmail.com>
+ *
+ * @internal
+ */
+final class TagResolver
+{
+    private $tags = array();
+
+    public function addTag(TagInterface $tag, $priority = 0)
+    {
+        if (!isset($this->tags[$priority])) {
+            $this->tags[$priority] = array();
+            krsort($this->tags);
+        }
+
+        $this->tags[$priority][] = $tag;
+    }
+
+    public function resolve($value, $tag)
+    {
+        $tag = $this->prefixTag($tag);
+        foreach ($this->tags as $registeredTags) {
+            foreach ($registeredTags as $registeredTag) {
+                if ($registeredTag->supports($value, $tag)) {
+                    return $registeredTag->construct($value, $tag);
+                }
+            }
+        }
+
+        throw new UnsupportedTagException(sprintf('Tag "%s" used with value "%s" is not supported', $tag, $value));
+    }
+
+    private function prefixTag($tag)
+    {
+        if ($tag && '!' === $tag[0]) {
```

Comment:
Is $tag a bool? If you validate $tag and it is a string I think it makes more sense to use any of the available methods just for improved readability.